### ISSUE TITLE
SMM intrinsic motivation signs

### ISSUE THREAD

**Issue by AOS55** [rll-research/url_benchmark#29]
Hey,

Not sure if anyone can clarify just wanted to check on signs with [intrinsic reward for SMM](https://github.com/rll-research/url_benchmark/blob/bb98f0c6d78b3c467fb5a9fa5bbba3b7c0250397/agent/smm.py#L263)

`intr_reward = pred_log_ratios + self.latent_ent_coef * h_z + self.latent_cond_ent_coef * h_z_s.detach()`

The [original paper](https://arxiv.org/pdf/1906.05274.pdf) in equation 3 has:

`r_z(s) = log(p*(s)) - log(rho_pi(s|z)) + log(p(z|s)) - log(p(z))`

Why do we add the `log(rho_pi(s|z)) == pred_log_ratios`  and `log(p(z)) == self.latent_ent_coef` and not subtract them as in equation 3, sorry if this is obvious 😄

**Comment by chhas** [comment]
I'm also wondering about the signs of the terms within SMM's intrinsic reward.
Regarding `pred_log_ratios`, I noticed that the [VAE](https://github.com/RLAgent/state-marginal-matching/blob/master/rlkit/density_models/vae_density.py#L72) of the original SMM implementation returns the negated `log_prob` (= `h_s_z`) value.
And within the [intrinsic reward](https://github.com/RLAgent/state-marginal-matching/blob/master/rlkit/torch/smm/smm_hook.py#L141) it is negated again.
Hence, [URLB's intrinsic reward](https://github.com/rll-research/url_benchmark/blob/main/agent/smm.py#L263) might be correct w.r.t. the sign of `h_s_z` because [URLB's VAE](https://github.com/rll-research/url_benchmark/blob/main/agent/smm.py#L66) does not negate the `log_prob` in the first place.
